src/VisualStudio/Core/Def/Implementation/ProjectSystem/LinkedFileUtilities.cs

```diff
@@ -152,5 +152,23 @@ private string GetActiveIntellisenseProjectContextInternal(IVsHierarchy hierarch
                 ? intellisenseProjectName as string
                 : null;
         }
+
+        public static bool TryGetItemIdInSharedHierarchy(IVsHierarchy hierarchy, uint itemId, IVsHierarchy sharedHierarchy, out uint itemIdInSharedHierarchy)
+        {
```

Comment:
Can you move the impl of this method to an instance method and AssertIsForeground()? Itemids are only guaranteed to be consistent as long as the consumer maintains ownership of the UI thread.